src/Orleans/Streams/PersistentStreams/PersistentStreamProvider.cs

```diff
@@ -60,11 +60,18 @@ public GrainFactoryServiceProvider(IStreamProviderRuntime providerRuntime)
             }
             public object GetService(Type serviceType)
             {
-                if (serviceType == typeof (GrainFactory))
+                var service = providerRuntime?.ServiceProvider?.GetService(serviceType);
```

Comment:
can the provider runtime be null because if so, getting the grain factory from it won't work.